File: packages/frame_ble/examples/stay_awake_false.py
import asyncio
from frame_ble import FrameBle

async def main():
    frame = FrameBle()

    try:
        await frame.connect()

        # stop any application, if running, so we can send lua commands
        await frame.send_break_signal()

        # Restore normal behavior that Frame turns off when placed in the charging cradle (and puts it to sleep now)
        await frame.send_lua("frame.stay_awake(false);print(0)", await_print=True)
        await frame.send_lua("frame.sleep()", await_print=False)
        print("Frame will switch off when placed in the charging cradle, and will be put to sleep now (tap to wake)")

        await frame.disconnect()

    except Exception as e:
        print(f"Not connected to Frame: {e}")
        return

if __name__ == "__main__":
    asyncio.run(main())

File: webbluetooth/packages/brilliant-ble/example/stay-awake-false.js
import { FrameBle } from 'frame-ble';

export async function run() {
  const frameBle = new FrameBle();

  // Connect to Frame
  const deviceId = await frameBle.connect();

  // Send a break signal to Frame in case it is in a loop/main.lua
  await frameBle.sendBreakSignal();

  // Restore normal behavior that Frame turns off when placed in the charging cradle (and puts it to sleep now)
  await frameBle.sendLua("frame.stay_awake(false);print(0)", {awaitPrint: true})
  await frameBle.sendLua("frame.sleep()")
  console.log("Frame will switch off when placed in the charging cradle, and will be put to sleep now (tap to wake)")

  // Disconnect from Frame
  await frameBle.disconnect();
};
